code/modules/projectiles/gun_helpers.dm

```diff
@@ -312,23 +310,9 @@ should be alright.
 		to_chat(user, "<span class='warning'>You need to disable overcharge on [src]!</span>")
 		return
 
-	//Checks if they can attach the thing in the first place, like with fixed attachments.
-	var/can_attach = 1
-	switch(attachment.slot)
-		if(ATTACHMENT_SLOT_RAIL)
-			if(rail && !(rail.flags_attach_features & ATTACH_REMOVABLE))
-				can_attach = FALSE
-		if(ATTACHMENT_SLOT_MUZZLE)
-			if(muzzle && !(muzzle.flags_attach_features & ATTACH_REMOVABLE))
-				can_attach = FALSE
-		if(ATTACHMENT_SLOT_UNDER)
-			if(under && !(under.flags_attach_features & ATTACH_REMOVABLE))
-				can_attach = FALSE
-		if(ATTACHMENT_SLOT_STOCK)
-			if(stock && !(stock.flags_attach_features & ATTACH_REMOVABLE))
-				can_attach = FALSE
-
-	if(!can_attach)
+	//Checks if there is any unremovable attachment on our slot, if true, return.
+	var/obj/item/attachable/currently_in_slot = LAZYACCESS(attachments, attachment.slot)
+	if(currently_in_slot && !(currently_in_slot.flags_attach_features & ATTACH_REMOVABLE))
```

Comment:
why not ?.

Reply:
i banana myself when doing ! and ?.

Reply:
if(null && !(null.var & flag)) returns false

if(!null?.var & flag) returns true

Reply:
That's not what I meant but whatever